platform/packages/finance/src/price/base/mod.rs

```diff
@@ -147,7 +147,7 @@ where
         price
             .quote()
             .try_into()
-            .and_then(|amount_quote| Self::new_checked(price.base().clone(), amount_quote))
+            .and_then(|amount_quote| Self::new_checked(price.base(), amount_quote))
```

Comment:
no need to change the `fn base()` to solve the problem. Review the function again to determine how to get ownership of `price.base`